src/presentation/components/common/ImageUpload/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { forwardRef, useState } from 'react';
+
+import { resizeImage } from '@utils/image';
+import { StImageUpload, StThumbnail } from './style';
+
+interface ImageUploadProps {
+  file: File | null | undefined;
+  setFile: (image: File) => void;
+  defaultThumbnail?: string;
+  style: {
+    width: string;
+    height: string;
+    borderRadius?: string;
+  };
+  children: React.ReactElement | string;
+  onClickInput: () => void;
+  openBottomSheet: () => void;
+  closeBottomSheet: () => void;
+  ref?: React.ForwardedRef<HTMLInputElement>;
+}
+const ImageUpload = forwardRef<HTMLInputElement, ImageUploadProps>((props, ref) => {
+  const {
+    children: emptyImage,
+    style,
+    openBottomSheet,
+    closeBottomSheet,
+    onClickInput,
+    file,
+    setFile,
+    defaultThumbnail = '',
+  } = props;
+  const [thumbnail, setThumbnail] = useState(defaultThumbnail);
+
+  const clickImageUpload = () => {
+    if (file instanceof File || (defaultThumbnail.length && file === undefined)) {
+      openBottomSheet();
+    } else {
+      onClickInput();
+    }
+  };
+
+  const handleFileInput = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.preventDefault();
+    if (e.target.files !== null && e.target.files.length > 0) {
```

Comment:
```suggestion
    if (e.target.files?.length > 0) {
```
이렇게 수정하는 게 좋을 것 같습니다 !

Reply:
넵 감사합니다!!

Reply:
아이고 타입내로잉 해줬던거였어요 `e.target.files !== null` 이렇게 명시를 해야 밑에서 e.target.files를 쓸 수 있어요 `HTMLInputElement.files: FileList | null` 이게 `e.target.files`의 타입입니다!
그래서 그대로 두겠습니다!